pkg/middleware/basic_session.go

```diff
@@ -61,7 +61,14 @@ func getBasicSession(validator basic.Validator, req *http.Request) (*sessionsapi
 
 	if validator.Validate(user, password) {
 		logger.PrintAuthf(user, req, logger.AuthSuccess, "Authenticated via basic auth and HTpasswd File")
-		return &sessionsapi.SessionState{User: user}, nil
+
+		session := &sessionsapi.SessionState{User: user}
+
+		if len(sessionGroups) > 0 {
```

Comment:
I don't think this is necessary - I think you can just set the `Groups` above, if the slice is nil or empty, it aligns with what the default struct would've made for `Groups` anyway had you not set it here.

Reply:
This was me being lazy :laughing: I will do this and fix the tests to suit.